File: src/Application/ClientApp/app/models/chat.model.ts
import { ChatMessage } from './chat-message.model';

export class Chat {
    id: number;
    createdDate: Date;
    timeToCloseChat: Date;
    sessionOrUserId: number;
    isUserAuthorized: boolean;
    userEmail: string;
    isChatEnded: boolean;
    shouldBeClosed: boolean = false;
    chatMessages: Array<ChatMessage> = new Array();
}

File: src/Entities/Models/ChatModels/ChatModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Application.EntitiesModels.Models.ChatModels
{
    public class ChatModel
    {
        public int Id { get; set; }

        public DateTimeOffset CreatedDate { get; set; }

        public DateTimeOffset TimeToCloseChat { get; set; }

        public string SessionOrUserId { get; set; }

        public bool IsUserAuthorized { get; set; }

        public string UserNameOrEmail { get; set; }

        public bool IsChatEnded { get; set; }

        public Guid ChatGUID { get; set; }

        public List<ChatMessageModel> ChatMessages { get; set; }

        public ChatModel()
        {
            ChatMessages = new List<ChatMessageModel>();
        }
    }
}
